File: DONGUK/프로그래머스_WEEK_14/기지국 설치.py
from collections import deque
import math

def solution(n, stations, w):
    answer = 0
    
    stations = deque(stations)
    wide = 2 * w + 1
    start = 1
    while stations:
        now = stations.popleft()
        end = now - w
        
        if start < end:
            before = end - start
            answer += math.ceil(before / wide)
            
        start = now + w + 1
    
    if start <= n:
        before = n - (now + w)
        answer += math.ceil(before / wide)
        
    return answer

File: JUNHEE/프로그래머스_옛날문제/기지국 설치.cpp
#include <iostream>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n, vector<int> stations, int w)
{
    int answer = 0;
    int range = 2 * w + 1;
    int start = 1;
    for(int i = 0; i < stations.size(); i++){
        int temp_start = stations[i] - w;
        if(temp_start <= start){
            start = stations[i] + w + 1;       
            continue;
        }
        answer += (temp_start - start) / range;
        if((temp_start - start) % range != 0) answer++;
        start = stations[i] + w + 1;
    }
    if(start - 1 < n){
        answer += (n + 1 - start) / range;
        if((n + 1 - start) % range != 0) answer++;        

    }

    return answer;
}
